test/unit/argument_parser/format_parse_validators_test.cpp

```diff
@@ -50,6 +53,36 @@ TEST(validator_test, fullfill_concept)
     EXPECT_TRUE(validator_concept<integral_range_validator<std::vector<int>>>);
     EXPECT_TRUE(validator_concept<value_list_validator<std::vector<int>>>);
     EXPECT_TRUE(validator_concept<file_ext_validator>);
+    EXPECT_TRUE(validator_concept<file_existance_validator>);
+}
+
+TEST(validator_test, no_file)
+{
+    filesystem::path p{"./sandbox.fasta"};
+    std::string s{"./stonebox.fasta"};
+    file_existance_validator my_validator{};
+    EXPECT_THROW(my_validator(p), parser_invalid_argument);
+    EXPECT_THROW(my_validator(s), parser_invalid_argument);
+
+    filesystem::path file_in_path;
+    std::vector<filesystem::path> path_vector;
+
+     // option
+     const char * argv[] = {"./argument_parser_test", "-i", "./sandbox.fasta"};
+     argument_parser parser("test_parser", 3, argv);
+     parser.add_option(file_in_path, 'i', "int-option", "desc",
+                       option_spec::DEFAULT, file_existance_validator());
+
+     EXPECT_THROW(parser.parse(), parser_invalid_argument);
+}
+
+TEST(validator_test, file_exists)
+{
+    test::tmp_filename tmp_file_name{""};
```

Comment:
The vary last things I think:

Can you give it a name instead of an empty string? 
and then I think you can remove the for loop because there will be only this one file in the directory.

Can you also add an argument_parser example here ?